src/appleseed/foundation/image/genericprogressiveimagefilereader.cpp

```diff
@@ -326,4 +342,100 @@ Tile* GenericProgressiveImageFileReader::read_tile(
     }
 }
 
+void GenericProgressiveImageFileReader::read_tile(
+    const size_t        tile_x,
+    const size_t        tile_y,
+    Tile*               output_tile)
+{
+    assert(is_open());
```

Comment:
Don't duplicate the whole function. I suspect that one function can easily used the other one.